android-components/components/browser/state/src/main/java/mozilla/components/browser/state/state/BrowserState.kt

```diff
@@ -54,4 +55,5 @@ data class BrowserState(
     val showExtensionsProcessDisabledPrompt: Boolean = false,
     val extensionsProcessDisabled: Boolean = false,
     val awesomeBarState: AwesomeBarState = AwesomeBarState(),
+    val translationEngine: TranslationsBrowserState = TranslationsBrowserState(),
```

Comment:
```suggestion
    val translationsEngineState: TranslationsEngineState = TranslationsEngineState(),
```
wdyt?

Reply:
Ohh, should have mentioned, the name of the class that the engine reports the tab state on is  `TranslationEngineState` (singular). 

Currently  have:
* [TranslationsState](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/browser/state/src/main/java/mozilla/components/browser/state/state/TranslationsState.kt) - This is the session one.
* [TranslationEngineState](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/translate/TranslationEngineState.kt) - This is information GeckoView reports about the session that is set on the session state.

For this one, I really struggled to find a new name:

`TranslationsBrowserEngineState`? `TranslationsAppState`? `TranslationsEngineStatus`? I keep struggling on this one to avoid a naming collision or something that makes sense.

Reply:
Maybe `TranslationsState` should become `TabTanslationsState` or `TranslationsSessionState`. Otherwise, `TranslationsBrowState` is fine.

Reply:
Okay, I'll leave `TranslationsBrowserState` for now, thanks for taking a look!

Agreed, I'll add a code health bug to refactor `TranslationsState` -> `TranslationsSessionState`. I think that'll help keep things more clear. Filed the session refactor naming to bug [1880352](https://bugzilla.mozilla.org/show_bug.cgi?id=1880352).